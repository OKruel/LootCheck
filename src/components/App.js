import React, { Component } from 'react'
import Routes from '../routes/routes'

class App extends Component {
    render() {
        return (
            <div>
                <Routes />
                <hr></hr>
                <div>Powered by <a target='_blank' rel="noopener noreferrer" href='https://www.coindesk.com/price'>Coindesk</a></div>
            </div>
        )
    }
}

export default App