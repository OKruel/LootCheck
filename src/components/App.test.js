//! Imports from React
import React from 'react'
//! Imports from Enzyme
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallow } from 'enzyme'
//! Imports from Main Files
import App from './App'

Enzyme.configure({ adapter: new Adapter() })

const app = shallow(<App />)

describe('App', () => {
    it('renders propely', () => {
        expect(App).toMatchSnapshot()
    })

    it('contains a Routes component', () => {
        // console.log(app.debug())
        expect(app.find('Routes').exists()).toBe(true)
    })

    it('contains a link to the coindesk price page', () => {
        expect(app.find('a').props().href).toBe('https://www.coindesk.com/price')
    })


})
