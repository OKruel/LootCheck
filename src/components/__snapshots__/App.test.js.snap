// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders propely 1`] = `[Function]`;
